python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -225,6 +268,26 @@ def is_json_serializable(value):
         return False
 
 
+def output_runtime_type_dict_conversion(runtime_type):
```

Comment:
Agreed what you said in our meeting @puma314 that an Enum would make this more clear. I also do think that you should experiment with consolidating the 'scalar' and 'any' use cases into a single one and see how that feels. I also think the names "output_def_str_dict" and "input_def_str_dict" could be better. Right now it maps input/output name to the serialization enum. Maybe just name it reflect that?